ddev/pyproject.toml

```diff
@@ -56,11 +56,49 @@ line-length = 120
 skip-string-normalization = true
 target-version = ["py38"]
 
-[tool.isort]
-default_section = "THIRDPARTY"
-force_grid_wrap = 0
-include_trailing_comma = true
-known_first_party = ["ddev"]
-line_length = 120
-multi_line_output = 3
-use_parentheses = true
+[tool.ruff]
+exclude = [
+    ".eggs",
+    ".tox",
+    "build",
+    "compat.py",
+    "__init__.py",
+    "**/datadog_checks/dev/tooling/templates/*",
+    "**/datadog_checks/*/vendor/*",
+]
+target-version = "py37"
+line-length = 120
+select = [
```

Comment:
Should we use [extend-select](https://beta.ruff.rs/docs/settings/#extend-select) here? Could we also add a comment to explain why we chose this particular set of letters? Similar to the comments we have for `ignore` items.

Reply:
I wanted to try doing this in the beginning, but then it was initially suggested that I port everything over from [hatch](https://github.com/pypa/hatch/blob/master/src/hatch/template/files_default.py#L186), but that wasn't possible due to us still supporting PY2. 

What I ended up doing was porting over all the setting we had with Flake8 instead:
https://github.com/DataDog/integrations-core/blob/master/.flake8#L3

There's no comments in there in regards to reasoning behind these rules. Each letter/ruleset also contains ~10-20 rules each:
https://beta.ruff.rs/docs/rules/#flake8-bugbear-b

Not sure if it's possible to annotate why we select or track down why we used these for flake8 at this point.

Reply:
Discussed and resolved offline